ngi_reports/utils/statusdb.py

```diff
@@ -65,38 +79,111 @@ def get_project_flowcell(self, project_id, open_date="2015-01-01", date_format="
             open_date = datetime.strptime("2015-01-01", "%Y-%m-%d")
 
         project_flowcells = {}
-        date_sorted_fcs = sorted(list(self.proj_list.keys()), key=lambda k: datetime.strptime(k.split('_')[0], "%y%m%d"), reverse=True)
-        for fc in date_sorted_fcs:
-            fc_date, fc_name = fc.split('_')
-            if datetime.strptime(fc_date,'%y%m%d') < open_date:
-                break
-            if project_id in self.proj_list[fc] and fc_name not in project_flowcells.keys():
-                project_flowcells[fc_name] = {'name':fc_name,'run_name':fc, 'date':fc_date, 'db':self.db.name}
-
+        if type(self) is NanoporeRunConnection:
+            found_fcs = []
+            for k in self.proj_list.keys():
+                try:
+                    date = datetime.strptime(k.split("_")[0], "%Y%m%d")
+                    found_fcs.append(k)
+                except ValueError:
+                    continue  # TODO: Revert this, it shouldn't be needed i production
+            date_sorted_fcs = sorted(
+                found_fcs,
+                key=lambda k: datetime.strptime(k.split("_")[0], "%Y%m%d"),
+                reverse=True,
+            )
+            for fc in date_sorted_fcs:
+                fc_date, fc_time, position, fc_name, fc_hash = fc.split(
+                    "_"
+                )  # 20220721_1216_1G_PAM62368_3ae8de85
+                if datetime.strptime(fc_date, "%Y%m%d") < open_date:
+                    break
```

Comment:
Do we need to keep this in ONT? Are there projs/runs before 2015 we need to steer clear of?

Reply:
Since the list is sorted, I guess all FCs that has open date dates from before the limit would not be needed?